zerver/lib/queue.py

```diff
@@ -308,8 +308,9 @@ def queue_json_publish(queue_name, event, processor):
             processor(event)
 
 def retry_event(queue_name, event, failure_processor):
-    # type: (str, Dict[str, Any], Callable[[Dict[str, Any]], None]) -> None
+    # type: (str, Mapping[str, Any], Callable[[Mapping[str, Any]], None]) -> None
     assert 'failed_tries' in event
+    event = cast(Mapping[str, Any], event)
```

Comment:
The `Mapping` type isn't mutable, so I think this is invalid from a typing perspective.  I think we should replace this piece of change with a larger preparatory commit that switches all the queue processors from expecting a `Mapping` to expecting a `Dict`, because that's what they're actually passed, and code like this will want to mutate it.

Reply:
Yeah.